tasks/main.yml

```diff
@@ -1 +1,60 @@
 # role tasks
+
+- name: Check localectl status before
+  command: localectl status
+  register: localectl
+  changed_when: "localectl.rc != 0"
+
+- name: set the localectl before fact
+  set_fact:
+    localectl_before: "{{ localectl.stdout_lines | join ('\n') }}"
+
+- name: Set the language locale
+  command: localectl set-locale "{{ localectl_system_locale}}"
+  register: locale
+  changed_when: "locale.rc != 0"
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+
+- name: Set the keymap
+  command: >
+    localectl set-keymap --no-convert "{{ localectl_vc_keymap }}"
+  register: keymap
+  changed_when: "keymap.rc != 0"
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+
+- name: Set the X11keymap
+  command: >
+    localectl set-x11-keymap --no-convert "{{ localectl_x11_layout }}"
+  register: layout
+  changed_when: "layout.rc != 0"
+  become: "{{ localectl_become }}"
+  become_user: "{{ localectl_become_user }}"
+
+- name: Check localectl status after
+  command: localectl status
+  register: localectl_after
+  changed_when: "localectl.rc != 0"
+
+- name: set the localectl after fact
+  set_fact:
+    locale_ctl_after: "{{ localectl_after.stdout_lines | join ('\n') }}"
```

Comment:
`stdout_lines` is `stdout` split on newlines. Rather than joining `stdout_lines` with newlines, just use `stdout`. However, working with stdout_lines is likely the better option here, due to parsing issues with how this fact is used described in comments below.

This fact is unnecessary, the registered outputs of each "set" task can be compared with the "before" values in `changed_when`/`failed_when` to report whether a change was made, or if the change failed to apply, respectively.